### PR TITLE
Fixing join game#29

### PR DESCRIPTION
**Pull request by davmmeyer**
After 3 hours spent with Git updates... I have finally corrected the errors. This push has already been merged with Master. Please note that the .gitignore file shouldn't cause any issues.

What has changed:
- Added create new game (with game name). 
- Added Join Game capability.

### CODE REVIEW

File: cos/static/modal.css [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
```
Comment (nadcock) [inline cos/static/modal.css:1]: I hate to do this to you with all the git problems you have had, but I don't think this is being used (bootstrap is taking care of this, and it is commented out in game.jinja2) and can be removed.

File: cos/static/modal.js [hunk 2]
```diff
@@ -35,4 +35,8 @@ window.onload = function() {
             'max-height':'100%'
         });
     });*/
-};
\ No newline at end of file
+};
+
+function dismiss_modal() {
```
Comment (nadcock) [inline cos/static/modal.js]: This function is empty and can be removed

File: cos/static/modal.css (hunk 1 above)
Comment (nadcock) [reply cos/static/modal.css:1]: Actually, it looks like this is still being used in landingpage.jinja2. We should probably at some point pull that out and replace it with the corrected bootstrap as was done in game.jinja2, but that can probably be done in a later ticket. What are your thoughts?

File: cos/templates/game.jinja2 [hunk 3]
```diff
@@ -1,33 +1,75 @@
...
       <!-- Bootstrap core CSS -->
-      <link href="//oss.maxcdn.com/libs/twitter-bootstrap/3.0.3/css/bootstrap.min.css" rel="stylesheet">
+      <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.2/css/bootstrap.min.css" integrity="sha384-PsH8R72JQ3SOdhVi3uxftmaW6Vc51MKb0q5P2rRUpPvrszuE4W1povHYgTpBfshb" crossorigin="anonymous">
+        <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.3/umd/popper.min.js" integrity="sha384-vFJXuSJphROIrBnz7yo7oB41mKfc8JzQZiCq4NCceLEaO4IHwicKwpJf9c9IpFgh" crossorigin="anonymous"></script>
+    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.2/js/bootstrap.min.js" integrity="sha384-alpBpkh1PFOepccYVYDB4do5UnbKysX5WZXm3XxPqe5iKTfUKjNkCk9SaVuEZflJ" crossorigin="anonymous"></script>
 
       <!-- Custom styles for this scaffold -->
-      <link href="{{request.static_url('cos:static/sandstone.css')}}" rel="stylesheet">
+{#      <link href="{{request.static_url('cos:static/sandstone.css')}}" rel="stylesheet">#}
```
Comment (nadcock) [inline cos/templates/game.jinja2]: These comments can be removed as they are not used

File: cos/templates/game.jinja2 [hunk 4]
```diff
@@ -1,33 +1,75 @@
 <!DOCTYPE html>
 <html lang="{{request.locale_name}}">
   <head>
+{#      <script type="text/javascript" src="{{request.static_url('cos:static/jquery-3.2.0.min.js')}}"></script>#}
```
Comment (nadcock) [inline cos/templates/game.jinja2]: This line can be removed as it is just commented out code that is no longer used.

File: cos/templates/game.jinja2 [hunk 5]
```diff
@@ -1,33 +1,75 @@
...
-      <link href="//oss.maxcdn.com/libs/twitter-bootstrap/3.0.3/css/bootstrap.min.css" rel="stylesheet">
+      <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.2/css/bootstrap.min.css" integrity="sha384-PsH8R72JQ3SOdhVi3uxftmaW6Vc51MKb0q5P2rRUpPvrszuE4W1povHYgTpBfshb" crossorigin="anonymous">
+        <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.3/umd/popper.min.js" integrity="sha384-vFJXuSJphROIrBnz7yo7oB41mKfc8JzQZiCq4NCceLEaO4IHwicKwpJf9c9IpFgh" crossorigin="anonymous"></script>
+    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.2/js/bootstrap.min.js" integrity="sha384-alpBpkh1PFOepccYVYDB4do5UnbKysX5WZXm3XxPqe5iKTfUKjNkCk9SaVuEZflJ" crossorigin="anonymous"></script>
 
       <!-- Custom styles for this scaffold -->
-      <link href="{{request.static_url('cos:static/sandstone.css')}}" rel="stylesheet">
+{#      <link href="{{request.static_url('cos:static/sandstone.css')}}" rel="stylesheet">#}
+{#      <link href="{{request.static_url('cos:static/modal.css')}}" rel="stylesheet">#}
```
Comment (nadcock) [inline cos/templates/game.jinja2]: Same here

**Review by nadcock — CHANGES_REQUESTED** [review]
@davmmeyer there are some minor changes (mainly with commented out code that we should try to limit merging into master). Also a question about if you want to use bootstrap in the landingpage template?

File: cos/static/modal.css (hunk 1 above)
Comment (davmmeyer) [reply cos/static/modal.css:1]: Yea, I have several custom functions in the modal.css (including modal size and dismiss on click outside) that would need to be ported. I think this is good for later review.

File: cos/static/modal.js (hunk 2 above)
Comment (davmmeyer) [reply cos/static/modal.js]: fixed

File: cos/templates/game.jinja2 (hunk 4 above)
Comment (davmmeyer) [reply cos/templates/game.jinja2]: fixed

File: cos/templates/game.jinja2 (hunk 3 above)
Comment (davmmeyer) [reply cos/templates/game.jinja2]: Fixed

File: cos/templates/game.jinja2 (hunk 5 above)
Comment (davmmeyer) [reply cos/templates/game.jinja2]: fixed

**Review by nadcock — APPROVED** [review]

**Review by mpolsgrove — APPROVED** [review]

**Pull request merged**